### ISSUE TITLE
Wording of m68k memory access limitations

### ISSUE THREAD

**Issue by jtsiomb** [fabiensanglard/cpsb#2]
Tiny issue with the wording of the 68k memory access limitations, not sure if I'd call it a mistake even, so I thought I'd report and let you decide if you want to reword it to make it clearer.

In section 2.5.1 you're saying: "Motorola's CPU will HALT while attempting to read/write memory not aligned on a 16-bit ( WORD ) boundary. This limitation is rooted all the way down to the CPU pins where there is no A0 line. Pins UDS and LDS indicate which of the high-byte or low-byte parts of a 16-bit WORD to access."

This is nearly correct. First of all it will not halt, it will raise an "address error" exception, which can be handled in software with the appropriate exception vector. Second and the gist of this issue, the 68k can access byte-aligned memory addresses, when reading/writing bytes. This works as you hinted, by putting the address bits 1-23 on the bus, and asserting LDS or UDS based on the A0 bit of the bus cycle address. This byte-aligned access, and especially with odd addresses (A0=1) are very common when communicating with 8-bit peripherals.

I've uploaded a couple of videos about the 68000 a few years ago, where in the second while explaining the design of my 68000 homebrew computer, I explained how I accidentally wired the bus backwards and was therefore forced to access all 8-bit peripherals on odd addresses by mistake :)  https://www.youtube.com/watch?v=jyiQshoC4ik  my previous video before that was a detailed demonstration of bus cycles of the 68000 on a breadboard, which you might or might not find useful: https://www.youtube.com/watch?v=rYkr1mFQ_50

**Comment by fabiensanglard** [comment]
1/ Thanks for your feedback (and super cool project by the way). I am going to change the wording from HALT to:

> throw an  "address error" exception

2/ I think readers will be able to figure out how `LDS` and `LDS` work. I will keep this comment in mind if other bring it up.

**Comment by jtsiomb** [comment]
Certainly, I just wanted to provide all the info, so you can decide what's the best way to word it. I wouldn't go so far as to explain the UDS/LDS mechanism, I'd just say something like "it allows only natively aligned memory accesses", or "while byte-sized accesses can be done on any address, word-sized accesses are only supported on word-aligned addresses", or something to that effect.

Btw I really like the book so far, very interesting. I'm looking forward to buy the print version. Also thank you for the SDK you put together. I'd like to give it a try as soon as I find some time.
